presto-common/src/main/java/com/facebook/presto/common/type/TypeSignature.java

```diff
@@ -609,6 +618,10 @@ else if (c == ',' && bracketLevel == 1) {
         return new TypeSignature(signature.substring(0, StandardTypes.ROW.length()), fields);
     }
 
+    private static void checkDuplicateAndAdd(Set<String> fieldNames, String fieldName)
+    {
+        verify(fieldNames.add(fieldName), "Duplicate field: " + fieldName);
```

Comment:
Can you help me understand why we need this change?

Reply:
This code is trying to handle the inconsistent behaviors in various statements about RowType provided by @mbasmanova. Listing as follows: 

```
presto> select typeof(cast(row(1, 2) as ROW(KEY1 VARCHAR, "key1" VARCHAR)));
                _col0
-------------------------------------
 row("key1" varchar, "key1" varchar)
(1 row)

presto> select cast(row(1, 2) as ROW(KEY1 VARCHAR, "key1" VARCHAR)).key1;
Query 20240209_085621_09358_9e2qz failed: Ambiguous field com.facebook.presto.common.type.RowType$Field@7b7cd13 in type row("key1" varchar, "key1" varchar)

presto> select cast(row(1, 2) as ROW(KEY1 VARCHAR, "key1" VARCHAR));
  _col0
----------
 {key1=2}
(1 row)
```

This code would check the keys in analyzing stage and consistently throw an exception if existing any duplicate ones.

Reply:
This shouldn't be a verify failure if the is something the user could send us in a query.  Maybe create a custom exception and handle it in Failures.toErrorCode(). This ultimately should be a semantic exception/user error. Otherwise we will ultimately wrap this in a Presto exception as a GENERIC_INTERNAL_ERROR.

Reply:
Sure, got it. I will fix it.